Microsoft.Azure.Cosmos/src/Query/Core/CoordinatorDistributionPlan/CoordinatorDistributionPlanDeserializer.cs

```
@@ -8,398 +8,724 @@ namespace Microsoft.Azure.Cosmos.Query.Core.CoordinatorDistributionPlan
-                    return DeserializeAggregateEnumerableExpression(token, serializer);
-                case Constants.Distinct:
-                    return DeserializeDistinctEnumerableExpression(token, serializer);
-                case Constants.GroupBy:
-                    return DeserializeGroupByEnumerableExpression(token, serializer);
-                case Constants.Flatten:
-                    return DeserializeFlattenEnumerableExpression(token, serializer);
-                case Constants.Input:
-                    return DeserializeInputEnumerableExpression(token, serializer);
-                case Constants.OrderBy:
-                    return DeserializeOrderByEnumerableExpression(token, serializer);
-                case Constants.ScalarAsEnumerable:
-                    return DeserializeScalarAsEnumerableExpression(token, serializer);
-                case Constants.Select:
-                    return DeserializeSelectEnumerableExpression(token, serializer);
-                case Constants.SelectMany:
-                    return DeserializeSelectManyExpression(token, serializer);
-                case Constants.Take:
-                    return DeserializeTakeEnumerableExpression(token, serializer);
-                case Constants.Where:
-                    return DeserializeWhereEnumerableExpression(token, serializer);
-                default:
-                    throw new NotSupportedException($"Unsupported ClientQLExpression kind: {kind}");
-            }
+            public const string UniqueId = "\"UniqueId\"";
+            public const string SortOrder = "\"SortOrder\"";
+            public const string Variable = "\"Variable\"";
+            public const string ObjectKind = "\"ObjectKind\"";
+            public const string Items = "\"Items\"";
         }
 
-        private static ClientQLAggregateEnumerableExpression DeserializeAggregateEnumerableExpression(JToken token, JsonSerializer serializer)
+        public struct Result
```

I am not able to see any value result is able to provide that following common pattern won't:
bool TryDeserialize<T>(CosmosObject obj, string property, out T clientQL)

---

In most cases, we want to fail (throw) when property is not found and fail fast (innermost function). In some cases you are permissive. For these two scenarios, you need 2 helper functions:
bool TryDeserialize<T>(CosmosObject obj, string property, out T clientQL);
T Deserialize<T>(CosmosObject obj, string property);

The second function just calls into first one and throws exception if it returns false.

Is this sufficient?